src/jsval.rs

```diff
@@ -179,20 +179,12 @@ pub fn ObjectOrNullValue(o: *mut JSObject) -> JSVal {
     }
 }
 
-#[cfg(target_pointer_width = "64")]
-#[inline(always)]
-pub fn PrivateValue(o: *const c_void) -> JSVal {
-    let ptrBits = o as usize as u64;
-    assert!((ptrBits & 1) == 0);
-    AsJSVal(ptrBits >> 1)
-}
-
-#[cfg(target_pointer_width = "32")]
 #[inline(always)]
 pub fn PrivateValue(o: *const c_void) -> JSVal {
     let ptrBits = o as usize as u64;
-    assert!((ptrBits & 1) == 0);
-    BuildJSVal(ValueTag::PRIVATE, ptrBits)
+    #[cfg(target_pointer_width = "64")]
+    assert_eq!(ptrBits & 0xFFFF000000000000, 0);
```

Comment:
`0xFFFF000000000001` to check the bottom bit?

Reply:
There is no bottom bit being shifted out anymore, that bitmask comes from SM itself.

https://searchfox.org/mozilla-central/source/js/public/Value.h#189

Reply:
TIL